dev/src/serializer.ts

```diff
@@ -118,7 +118,12 @@ export class Serializer {
     }
 
     if (typeof val === 'number') {
-      if (Number.isSafeInteger(val)) {
+      const isNegativeZero = val === 0 && 1 / val === -Infinity;
```

Comment:
Just for our future sanity, can we use the exact same check that we use on the Web SDK? https://github.com/firebase/firebase-js-sdk/blob/254137540a0e73d357a9a8cd27c5826108679f7b/packages/firestore/src/util/types.ts#L37

Otherwise this LGTM.

Reply:
Done :)